### DISCUSSION TITLE
Service inaccessible from the node it runs on

### DISCUSSION THREAD

**Discussion by ronen4822** [loxilb-io/loxilb#598] [Q&A] [answered]
Hey,

I'm currently working on deploying traefik with flannel on k3s, using in-cluster mode. The traefik service successfully gets assigned an IP by loxilb and is accessible from remote machines. However, I'm facing a problem with accessing services exposed via an ingress when trying to access them from within the same node or even when attempting to curl from the node that hosts the pod. I keep receiving a "connection refused" error.

I've also checked the DNS resolution on the node and everything seems fine.
I've also tried accessing the node port which seems to work  (just once, followed my 302 status codes).
It seems like I might be overlooking something trivial, but I'm struggling to identify the root cause of the failure.

The kube-loxilb runs with the following args:
        - --externalCIDR=NODE_EXTERNAL_IP/32
        - --setRoles=0.0.0.0

Any assistance on this matter would be greatly appreciated.

Thanks!

**Comment by TrekkieCoder** [comment]
Currently, the VIP is available via serviceLB (inside K8s node) if the VIP subnet is not local to the host and (hypothetically) reachable via some route. For example host is (eth0) 192.168.10.1/24 and VIP is 10.10.10.1 and say a default route exists via 192.168.10.254(gw). it might be too complex  but loxilb in its default mode expects external traffic (reverse proxy).  But still some users have worked with the above scheme. 

Meanwhile, we are working on a permanent solution as there seems to be a genuine requirement for this.

**Comment by UltraInstinct14** [comment, marked as answer]
The support for connecting to VIP from within the node for in-cluster mode is now available (although in beta). You need to redeploy loxilb daemonset (delete and reapply) using this [yaml](https://github.com/loxilb-io/kube-loxilb/blob/main/manifest/in-cluster/loxilb-localvip.yaml) . No need for any change in kube-loxilb.
